modules/backup-gcs/client.go

```diff
@@ -273,35 +239,35 @@ func (g *gcsClient) WriteToFile(ctx context.Context, backupID, key, destPath str
 	}()
 
 	// create reader
-	object := g.makeObjectName(backupID, key)
+	object := g.makeObjectName(overridePath, []string{backupID, key})
 	rc, err := bucket.Object(object).NewReader(ctx)
 	if err != nil {
-		return fmt.Errorf("find object %q: %w", object, err)
+		return fmt.Errorf("create reader for writetofile %q: %w", object, err)
 	}
 	defer rc.Close()
 
 	// transfer content to the file
 	if _, err := io.Copy(file, rc); err != nil {
-		return fmt.Errorf("io.Copy:%q %q: %w", destPath, object, err)
+		return fmt.Errorf("io.Copy for writetofile:%q %q: %w", destPath, object, err)
 	}
 	closeAndRemove = false
 	if err = file.Close(); err != nil {
-		return fmt.Errorf("f.Close %q: %w", destPath, err)
+		return fmt.Errorf("f.Close for writetofile %q: %w", destPath, err)
 	}
 
 	return nil
 }
 
-func (g *gcsClient) Write(ctx context.Context, backupID, key string, r io.ReadCloser) (int64, error) {
+func (g *gcsClient) Write(ctx context.Context, backupID, key, overrideBucket, overridePath string, r io.ReadCloser) (int64, error) {
 	defer r.Close()
 
-	bucket, err := g.findBucket(ctx)
+	bucket, err := g.findBucket(ctx, overrideBucket)
 	if err != nil {
-		return 0, fmt.Errorf("find bucket: %w", err)
+		return 0, fmt.Errorf("gcs write: %w", err)
```

Comment:
suggestion: maybe `write: find bucket: %w` message here would be more appropriate?

Reply:
done